ml_tools/model/state.py

```diff
@@ -332,12 +335,14 @@ def chunkify(states: List[State], chunk_size: int):
             with ProcessPoolExecutor(max_workers=num_procs) as executor:
                 jobs = {executor.submit(State.perturb_state(perturbators, state)): chunk for chunk in chunks}
 
-                for i, job in enumerate(jobs):
-                    for state in job.result():
-                        state_data.append(state)
-                        if not silent:
-                            statusbar.update(i)
-                            i+=1
+                completed = 0
+                for job in as_completed(jobs):
+                    result = job.result()
+                    state_data.extend(result)
+                    if not silent:
+                        for _ in result:
+                            statusbar.update(completed)
+                            completed += 1
```

Comment:
It's a nitpick, but this and lines 239-246 could potentially be combined into their own little routine

Reply:
There's probably a better way to do statusbar in general for parallel stuff, but making this change here without a significant refactor is looking like more trouble than it's worth.

Reply:
ok, fair enough